ts/Series/ArcDiagram/ArcDiagramSeries.ts

```diff
@@ -0,0 +1,782 @@
+         * @since next
+         * @default undefined
+         * @product highcharts
+         * @apioption series.arcdiagram.linkRadius
+         */
+
+        /**
+         * The offset of an arc diagram nodes column in relation
+         * to the plotArea. The offset equal to 50% places nodes
+         * in the center of a chart.
+         *
+         * @type    {string}
+         * @since next
+         * @default undefined
+         * @product highcharts
+         * @apioption series.arcdiagram.offset
+         */
+
+        /**
+         * The global link weight. If not set, width will is calculated
+         * per link, depending on the weight value.
+         *
+         * @type    {number}
+         * @since next
+         * @default undefined
+         * @product highcharts
+         * @apioption series.arcdiagram.linkWeight
+         */
+
+        /**
+         * Whether nodes with different values should have the same size.
+         *
+         * @type    {boolean}
+         * @since next
+         * @default false
+         * @product highcharts
+         */
+        equalNodes: false,
+        keys: ['from', 'to', 'weight'],
+        /**
+         * Whether series should be placed on the other side of plotArea.
+         *
+         * @type    {boolean}
+         * @since next
+         * @default false
+         * @product highcharts
+         */
+        reversed: false,
+        /**
+         * Options for the data labels appearing on top of the nodes and links.
+         * For Arc diagram charts, data labels are visible for the nodes
+         * by default, but hidden for links. This is controlled
+         * by modifying the `nodeFormat`, and the `format` that applies to links
+         * and is an empty string by default.
+         *
+         * @declare Highcharts.SeriesArcDiagramDataLabelsOptionsObject
+         *
+         * @private
+         */
+        dataLabels: {
+            /**
+             * Options for a _link_ label text which should follow link
+             * connection. Border and background are disabled for a label that
+             * follows a path.
+             *
+             * **Note:** Only SVG-based renderer supports this option. Setting
+             * `useHTML` to true will disable this option.
+             *
+             * @extends plotOptions.arcdiagram.dataLabels.linkTextPath
+             * @since   next
+             */
+            linkTextPath: {
+                attributes: {
+                    startOffset: '25%'
+                }
+            }
+        },
+        /**
+         * @extends   plotOptions.series.marker
+         * @excluding enabled, enabledThreshold, height, radius, width
+         */
+        marker: {
+            symbol: 'circle',
+            /**
+             * In arc-diagram charts, the radius is overridden and determined
+             * based on the point's data value.
+             *
+             * @ignore-option
+             */
```

Comment:
Missing docs on how to control the size of the marker when `equalNodes` is true. It looks - from the code - to be possible to control it using `nodePadding`. But maybe we also should allow setting `marker.radius` so that it is possible to set it to a fixed value? For example if I want to make it the same height as my data labels.